### ISSUE TITLE
support python 3.15 `sentinel`

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3752]
https://peps.python.org/pep-0661/

`main.py`:

```py
MISSING = sentinel('MISSING')

def fn(value: int | MISSING = MISSING): ... # error: invalid-type-form
```

`ty.json`:

```json
{
    "environment": {
        "python-version": "3.15"
    },
    "rules": {
        "all": "error",
        "empty-body": "ignore"
    }
}
```

[Playground](https://play.ty.dev/da64d5e9-5c9d-416f-a824-8b16df5a3a71)

**Comment by MichaReiser** [comment]
See https://github.com/astral-sh/ruff/pull/25082 for why it was not added in the initial Sentinel support

**Comment by carljm** [comment]
I think this will just naturally happen without any changes in ty as typeshed is updated to reflect the existence of these names. The core support already exists; the OP example works with `from typing_extensions import Sentinel as sentinel`.

Closing since I don't think the remaining work here is on the ty side.

**Comment by AlexWaygood** [comment]
That's not true. We've bundled a version of typeshed for several weeks now that includes `builtins.sentinel`, but we still only support it in ty if it's imported from `typing_extensions`

**Comment by AlexWaygood** [comment]
https://github.com/astral-sh/ruff/pull/26262 demonstrates what changes in ty are needed. Unfortunately I was also incorrect, though -- this does require some upstream changes in typeshed as well as ty (and those upstream changes in typeshed are themselves blocked on a new `typing_extensions` release).
